### PR TITLE
satellite/audit: delete nodes from containment when segment has changed

### PR DESCRIPTION
**Pull request by mobyvb**
What: 
Delete a node from containment when a segment has been deleted or no longer contains the contained node inside audit Reverify.

Remove containment delete logic from metainfo.

Why:
We do not want to keep nodes in containment if they can no longer be audited for the initial share they were contained for (because they are not holding that share anymore for legitimate reasons)

We do not want to delete nodes from containment when they are in a segment being deleted, since they may have been contained for a different segment.

See more detailed discussion here: https://github.com/storj/storj/pull/3104#discussion_r327767953

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/metainfo/metainfo.go [hunk 1]
```diff
@@ -412,13 +412,6 @@ func (endpoint *Endpoint) DeleteSegmentOld(ctx context.Context, req *pb.SegmentD
 	}
 
 	if pointer.Type == pb.Pointer_REMOTE && pointer.Remote != nil {
-		for _, piece := range pointer.GetRemote().GetRemotePieces() {
-			_, err := endpoint.containment.Delete(ctx, piece.NodeId)
```
Comment (egonelbre) [inline satellite/metainfo/metainfo.go]: Does metainfo still require containment?

File: satellite/audit/verifier.go [hunk 2]
```diff
@@ -357,8 +375,13 @@ func (verifier *Verifier) Reverify(ctx context.Context, path storj.Path) (report
 				}
 			}
 			if !found {
+				// node is no longer in pointer, so remove from containment
```
Comment (egonelbre) [inline satellite/audit/verifier.go]: The question is that when the segment gets deleted for a node, should we 
1. remove it from containment,
2. pick another piece/segment, or
3. it doesn't matter.

File: satellite/metainfo/metainfo.go (hunk 1 above)
Comment (mobyvb) [reply satellite/metainfo/metainfo.go]: I don't think so. I will remove it.

File: satellite/audit/verifier.go (hunk 2 above)
Comment (mobyvb) [reply satellite/audit/verifier.go]: I think picking another piece/segment might be the ideal solution, but that is probably too complex of a change to do for now. If it is a bad node, it should end up in containment again anyway even if we remove the node from containment here.

File: satellite/metainfo/metainfo.go (hunk 1 above)
Comment (mobyvb) [reply satellite/metainfo/metainfo.go]: Done

**Pull request merged**